server.port=8083
spring.application.name=cloud-eureka-consumer

eureka.client.service-url.defaultZone=http://localhost:8081/eureka/
# 开启feign的断路器功能, 出现异常服务异常(指网络异常等非逻辑异常)
feign.hystrix.enabled=true
# 连接超时时间
ribbon.ConnectionTimeout=1000
# 读取数据超时时间, 时间段内没有响应则根据重试次数规则发起重试.  重试机制完成后, 还没有返回结果则触发断路器, 开始服务降级逻辑
ribbon.ReadTimeout=3000
# 所有HTTP操作都允许重试机制, 默认情况下使用的是GET请求或者该参数配置的为true则进行重试
ribbon.OkToRetryOnAllOperations=false
# 服务重试和hystrix没有关系, 即使hystrix做了服务降级处理, 只要重试机制的条件满足, 仍然会重试.
# 所以不要想着通过设置ribbon的超时时间大于hystrix的超时时间来防止重试机制触发
# 其他服务最大重试次数
ribbon.MaxAutoRetriesNextServer=0
# 当前服务最大重试次数
ribbon.MaxAutoRetries=0

# 关闭重试机制, 该配置无效...
#spring.cloud.loadbalancer.retry.enabled=false

hystrix.command.default.execution.timeout.enabled=true
# 时间段内没有响应, 则服务中断,回调服务降级逻辑
hystrix.command.default.execution.isolation.thread.timeoutInMilliseconds=7000
